third_party/blink/renderer/modules/cobalt/h5vcc_runtime/h_5_vcc_runtime.h

```diff
@@ -77,13 +74,19 @@ class MODULES_EXPORT H5vccRuntime final
 
  private:
   void MaybeFireDeepLinkEvent(const String&);
-  void EnsureReceiverIsBound();
-  void RemoteAddListener();
+  void EnsureRemoteIsBound();
+
+  void MaybeRegisterMojoListener();
+  void MaybeUnregisterMojoListener();
+
   HeapMojoRemote<h5vcc_runtime::mojom::blink::H5vccRuntime>
       remote_h5vcc_runtime_;
-  HeapMojoReceiver<h5vcc_runtime::mojom::blink::DeepLinkListener, H5vccRuntime>
-      receiver_;
   String initial_deep_link_;
+
+  // Track if AddListener was called
+  bool listener_registered_ = false;
```

Comment:
Instead of tracking this ourself, can't we use `listener_receiver_.is_bound()`? https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/mojo/heap_mojo_receiver.h;l=46?q=HeapMojoReceiver&ss=chromium%2Fchromium%2Fsrc